src/main/org/firebirdsql/jdbc/AbstractResultSet.java

```diff
@@ -73,6 +75,7 @@
     private SQLWarning firstWarning;
 
     private final FBField[] fields;
+    private List<FBField> closeableFields = null;
```

Comment:
Consider always initializing this with an `ArrayList` with `new ArrayList(0)` or `new ArrayList()`, that removes the need for null-checks.

And otherwise, remove the explicit `null` assignment.